daisy/machinetype.go

```diff
@@ -18,40 +18,29 @@ import (
 	"fmt"
 	"regexp"
 
-	"github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
 )
 
 var machineTypeURLRegex = regexp.MustCompile(fmt.Sprintf(`^(projects/(?P<project>%[1]s)/)?zones/(?P<zone>%[2]s)/machineTypes/(?P<machinetype>%[2]s)$`, projectRgxStr, rfc1035))
 
-var machineTypeCache twoDResourceCache
-
-func machineTypeExists(client compute.Client, project, zone, machineType string) (bool, DError) {
-	machineTypeCache.mu.Lock()
-	defer machineTypeCache.mu.Unlock()
-	if machineTypeCache.exists == nil {
-		machineTypeCache.exists = map[string]map[string][]interface{}{}
-	}
-	if _, ok := machineTypeCache.exists[project]; !ok {
-		machineTypeCache.exists[project] = map[string][]interface{}{}
+func (w *Workflow) machineTypeExists(project, zone, machineType string) (bool, DError) {
+	predefinedMachineTypeExists, err := w.machineTypeCache.resourceExists(func(project, zone string, opts ...daisyCompute.ListCallOption) (interface{}, error) {
+		return w.ComputeClient.ListMachineTypes(project, zone)
+	}, project, zone, machineType)
+	if err != nil {
+		return predefinedMachineTypeExists, err
```

Comment:
Convention is to avoid populating other return values on error (this can cause very hard errors to debug):
return false, err